addons/stock/models/stock_warehouse.py

```diff
@@ -155,6 +163,12 @@ def create(self, vals_list):
 
         return warehouses
 
+    @api.model
+    def _warehouse_redirect_warning(self):
+        warehouse_action = self.env.ref('stock.action_warehouse_form')
+        msg = _('Please create a warehouse for this company.')
```

Comment:
I would add a UserError in case user is not Inventory Manager 
"Please contact your administrator blabla  ..." 
He cannot create warehouse anyway